boa_engine/src/builtins/error/mod.rs

```diff
@@ -43,6 +44,18 @@ pub(crate) use self::uri::UriError;
 
 use super::JsArgs;
 
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+pub enum ErrorKind {
+    Aggregate,
+    Error,
+    Eval,
+    Type,
+    Range,
+    Reference,
+    Syntax,
+    Uri,
+}
```

Comment:
I'll just relink to `JsNativeError`, since this is mostly internal. I'll also make it crate-only, just to ensure it is always set to the correct variant.